src/qualifed_name.rs

```diff
@@ -0,0 +1,244 @@
+use crate::{index::Index, typesystem::Dimension};
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum QualifiedNameElement<'idx> {
+    Name(&'idx str),
+    ArrayAccess(&'idx [Dimension]),
+}
+
+impl ToString for QualifiedNameElement<'_> {
+    fn to_string(&self) -> String {
+        match self {
+            QualifiedNameElement::Name(name) => name.to_string(),
+            QualifiedNameElement::ArrayAccess(_) => todo!(),
+        }
+    }
+}
+
+impl<'idx> From<&'idx str> for QualifiedNameElement<'idx> {
+    fn from(name: &'idx str) -> Self {
+        QualifiedNameElement::Name(name)
+    }
+}
+
+impl<'idx> From<&'idx [Dimension]> for QualifiedNameElement<'idx> {
+    fn from(dims: &'idx [Dimension]) -> Self {
+        QualifiedNameElement::ArrayAccess(dims)
+    }
+}
+
+#[derive(Debug, Default, Clone, PartialEq)]
+pub struct QualifiedName<'idx> {
```

Comment:
I'm also unsure regarding this name.
Whatever we call qualified_name in the index, means something different. 

If I get this right, this qualified name may be something like:
`PLC_PRG.fb1.fb2.x`

while the index would treat x's qualified name as: `FB2.x`

Reply:
not sure if we have QualifiedExpressions in plc-dev